### PR TITLE
Add support for hugepage on s390x

### PR DESCRIPTION
**Pull request by fbq815**
add 1M hugepage supported and host hugepage set for supporting s390x

ID:1936832

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
Test result:
JOB ID     : 764a020d3d57ad9bf2442c2254d33135ec9d2636
JOB LOG    : /home/bfu/kar/workspace/job-results/job-2021-03-09T07.21-764a020/job.log
 (1/1) Host_RHEL.m8.u4.product_rhel.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.s390x.io-github-autotest-qemu.boot_vm_in_hugepage.1M.s390-virtio: PASS (217.04 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/bfu/kar/workspace/job-results/job-2021-03-09T07.21-764a020/results.html
JOB TIME   : 217.75 s

**Comment by fbq815** [comment]
@PaulYuuu @MiriamDeng  Please help me review this patch

File: generic/tests/cfg/boot_vm_in_hugepage.cfg [hunk 1]
```diff
@@ -9,16 +9,24 @@
     extra_params += " -mem-path /mnt/kvm_hugepage"
     pre_command_noncritical = yes
     pre_command = "echo 3 > /proc/sys/vm/drop_caches"
+    s390x:
+        kvm_module_parameters = 'hpage=1'
     variants:
+        - 1M:
```
Comment (MiriamDeng) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg:15]: Is 1M huge page only supported on s390x ? If so, we need to set it "only s390x", thanks.

**Comment by MiriamDeng** [comment]
And also the commit log should be like this, feature name: xxxxx

File: generic/tests/cfg/boot_vm_in_hugepage.cfg (hunk 1 above)
Comment (PaulYuuu) [inline generic/tests/cfg/boot_vm_in_hugepage.cfg:15]: If you wanna add hardware restrictions for different huge page settings, you should consider all platforms.
So I think you can just add `- 1M` variant and pick it at your test loop.

Comment (fbq815) [reply generic/tests/cfg/boot_vm_in_hugepage.cfg:15]: I decided to add "only s390x" to avoid conflict with other architechture

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**